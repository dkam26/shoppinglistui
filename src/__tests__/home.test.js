import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import Login from '../components/home.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import ReactDOM from 'react-dom';
configure({ adapter: new Adapter() });
describe('login component', () => {
    const props={
        render:()=>{}
    }
    const wrapper =  shallow(<Login/>);
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should render correctly', () => {
        const output = shallow( <Login />);
        expect(shallowToJson(output)).toMatchSnapshot();
        });
         it('render the two inputs', () =>{
            
            expect(wrapper.find("#username")).toHaveLength(1);
             expect(wrapper.find("#password")).toHaveLength(1);
         
         });
    it('renders without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(<Login />, div);
        }); 
    it('should ve the following methods',()=>{
            wrapper.instance().render();
           
        });
})