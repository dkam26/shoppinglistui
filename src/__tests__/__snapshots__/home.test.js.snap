// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component should render correctly 1`] = `
<div
  className="login-form"
>
  <style>
    
      body &gt; div,
      body &gt; div &gt; div,
      body &gt; div &gt; div &gt; div.login-form {
        height: 100%;
      }
    
  </style>
  <Grid
    style={
      Object {
        "height": "100%",
      }
    }
    textAlign="center"
    verticalAlign="middle"
  >
    <GridColumn
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <Header
        as="h2"
        color="black"
        textAlign="center"
      >
        Shoppinglists-App
      </Header>
      <Form
        as="form"
        size="large"
      >
        <Segment
          stacked={true}
        >
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="user"
            iconPosition="left"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
          />
          <FormInput
            as={[Function]}
            control={[Function]}
            fluid={true}
            icon="lock"
            iconPosition="left"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <Button
            as="button"
            color="blue"
            fluid={true}
            onClick={[Function]}
            size="large"
          >
            Login
          </Button>
        </Segment>
      </Form>
      <_class
        options={Object {}}
      />
      <Message>
        New to us? 
        <a
          href="/register"
        >
          Sign Up
        </a>
      </Message>
    </GridColumn>
  </Grid>
</div>
`;
